bin/commands/init/vsam/index.ts

```diff
@@ -0,0 +1,166 @@
+  let vsam_volume='';
+  if (vsam_mode == "NONRLS") {
+    vsam_volume=zoweConfig.zowe.setup?.vsam?.volume;
+    if (!vsam_volume) {
+      common.printErrorAndExit(`Error ZWEL0157E: Zowe Caching Service VSAM data set Non-RLS volume (zowe.setup.vsam.volume) is not defined in Zowe YAML configuration file.`, undefined, 157);
+    }
+  }
+  let vsam_storageClass='';
+  if (vsam_mode == "RLS") {
+    vsam_storageClass=zoweConfig.zowe.setup?.vsam?.storageClass;
+    if (!vsam_storageClass) {
+      common.printErrorAndExit(`Error ZWEL0157E: Zowe Caching Service VSAM data set RLS storage class (zowe.setup.vsam.storageClass) is not defined in Zowe YAML configuration file.`, undefined, 157);
+    }
+  }
+  const vsam_name=zoweConfig.components['caching-service'].storage?.vsam?.name;
+  if (!vsam_name) {
+    common.printErrorAndExit(`Error ZWEL0157E: Zowe Caching Service VSAM data set name (components.caching-service.storage.vsam.name) is not defined in Zowe YAML configuration file.`, undefined, 157);
+  }
+
+  const jcl_existence=zosdataset.isDatasetExists(`${jcllib}(ZWECSVSM)`);
+  if (jcl_existence == true) {
+    if (allowOverwrite) {
+      common.printMessage(`Warning ZWEL0300W: ${jcllib}(ZWECSVSM) already exists. This data set member will be overwritten during configuration.`);
+    } else {
+      common.printMessage(`Warning ZWEL0301W: ${jcllib}(ZWECSVSM) already exists and will not be overwritten. For upgrades, you must use --allow-overwrite.`);
+    }
+  }
+
+  // VSAM cache cannot be overwritten, must delete manually
+  // FIXME: cat cannot be used to test VSAM data set
+  const vsam_existence=zosdataset.isDatasetExists(vsam_name);
+  if (vsam_existence == true) {
+    common.printErrorAndExit(`Error ZWEL0158E: ${vsam_name} already exists.`, undefined, 158);
+  }
+  if (allowOverwrite) {
+    // delete blindly and ignore errors
+    let result=zoslib.tsoCommand('delete', `'${vsam_name}'`);
+  }
+
+
+  if (jcl_existence == true && allowOverwrite != true) {
+    common.printMessage(`Skipped writing to ${jcllib}(ZWECSVSM). To write, you must use --allow-overwrite.`);
+  } else {
+    // prepare STCs
+    // ZWESLSTC
+    common.printMessage(`Modify ZWECSVSM`);
+    const replacer = new RegExp('\ ', 'g');
+    const tmpfile=fs.createTmpFile(`zwe ${std.getenv('ZWE_CLI_COMMANDS_LIST')}`.replace(replacer, '-'));
+    common.printDebug(`- Copy ${prefix}.${std.getenv('ZWE_PRIVATE_DS_SZWESAMP')}(ZWECSVSM) to ${tmpfile}`);
+
+    const theDataset = shell.execOutSync('sh', '-c', `cat "//'${prefix}.${std.getenv('ZWE_PRIVATE_DS_SZWESAMP')}(ZWECSVSM)'" 2>&1`);
+    if (theDataset.out && theDataset.rc == 0) {
+      common.printDebug(`  * Succeeded`);
+      common.printTrace(`  * Output:`);
+      common.printTrace(stringlib.paddingLeft(theDataset.out, "    "));
+
+      const tmpFileContent = theDataset.out.replace(/\n\/\/\s*SET MODE=.*\n/, `\n//         SET  MODE=${vsam_mode}\n`)
+                .replace(new RegExp('[(]\#dsname', 'g'), '(' + vsam_name)
```

Comment:
The shell version is replacing after the `//ALLOC` statement. This is simple way how to replace `#dsname`, `#volume` and `#storclas` in the `DEFINE CLUSTER` section only. It is searching for `(#variable` and replace it with `(value_of_the_variable`. It will keep `#variable` in the comment section unchanged. Feel free to modify it.